doc/howtos/rdtraining/E_unittest.rst

```diff
@@ -0,0 +1,212 @@
+A sub-build is greened if all the tests are passing and there are no errors/warnings logged.
+
+.. note::
+  * Robodoo doesn't care what the changes were. It will always run the same tests, meaning it
+    will install all the modules. This means something might not work if runbot greened but your
+    changes depend on something you don't depend on.
+  * The localization modules are not installed on runbot (except the generic one), some modules
+    with external dependencies can be excluded also.
+  * There is a nightly build running additional tests, like module operations, localization, single
+    module installs, multi-builds for nondeterministic bugs, etc.
+    These are not kept in the standard CI to shorten the time of execution.
+
+You can also login on a build built by runbot. There are 3 users usable: `admin`, `demo` and
+`portal`. The password is the same as the login. This is useful to quickly test things on different
+versions without having to build it locally. The full logs are also available; these are used for
+monitoring.
+
+Robodoo
+-------
+
+You will most likely have to gain a little bit more experience before having the rights to summon
+robodoo, but here are a few remarks anyways.
+
+Robodoo is the guy spamming the CI status as tags on your PRs, but he is also the guy that kindly
+integrates your commits on the main repositories.
+
+When the last batch is greened, the reviewer can ask robodoo to merge your PR (actually it is more
+a `rebase` than a `merge`). It will then go to the mergebot.
+
+
+Mergebot
+--------
+
+`Mergebot <https://mergebot.odoo.com>`__ is the last testing phase before merging a PR.
+
+It will take the commits in your branch not yet present on the target, stage it and rerun the tests
+one more time, including the enterprise version even if you are only changing something in
+community.
+
+Modules
+=======
+
+Because Odoo is modular, the tests need to be modular also. This means the modules are defined in
+the module that adds the functionality you add; and that tests cannot depend on functionality
+coming from modules your module doesn't depend on.
+
+If the behavior you want to test can be changed by the installation of another module, you need to
+ensure that the tag `at_install` is set; otherwise you can use the tag `post_install` to speed up
+the CI, and ensure it is not changed if it shouldn't.
+
+Writing a test
+==============
+
+**Reference**: the documentation related to this topic can be found in
+`Python unittest <https://docs.python.org/3/library/unittest.html>`__.
+
+**Reference**: the documentation related to this topic can be found in
+:ref:`Testing Odoo<reference/testing>`.
+
+Here are a few things to take into consideration before writing a test
+
+* The tests should be independent from the data currently in the database
+* Tests should not impact the database by leaving/changing residual data. This is usually done by
+  the test framework
```

Comment:
```suggestion
  the test framework. You must never call `cr.commit` in a test.
```

Reply:
This is not proper to tests, is it? Calling `cr.commit` will have undesired effects almost anywhere, shouldn't it be part of the guidelines?

Reply:
indeed, it should be called only in exceptional cases.